File: ru.nsu.fit.app/src/entities/Subject.java
package entities;

import java.util.List;

public class Subject {
    public String educationalProgram;
    public String subjectName;
    public String semesters;
    public String typeOfClass;
    public String frequency;
    public String teacher;
    public String amountOfGroups;
    public List<Teacher> teachers;

    public Subject(String educationalProgram, String subjectName,
                   String semesters, String typeOfClass, String frequency,
                   String teacher, String amountOfGroups) {
        this.educationalProgram = educationalProgram;
        this.subjectName = subjectName;
        this.semesters = semesters;
        this.typeOfClass = typeOfClass;
        this.frequency = frequency;
        this.teacher = teacher;
        this.amountOfGroups = amountOfGroups;
    }
}

File: PrologSolver/entities/Subject.py
class Subject:
    def __init__(self, id, specializationId, specialization, name, semesters,
                 typeOfClass, frequency, teacherId, amountOfGroups, generated):
        self.id = id
        self.specializationId = specializationId
        self.specialization = specialization
        self.name = name
        self.semesters = semesters
        self.typeOfClass = typeOfClass
        self.frequency = frequency
        self.teacherId = teacherId
        self.amountOfGroups = amountOfGroups
        self.generated = generated

    def getSemesters(self):
        tokens = self.semesters.split(",")

        return tokens
